src/app/hooks/keycloak.ts

```diff
@@ -13,18 +13,19 @@ const init = async (config: Config, getInsightsAccessToken: () => Promise<string
   return await initKeycloak(keycloakConfig, getInsightsAccessToken);
 }
 
-export const useAuth = (): Auth => {
+export const useAuth = (): Auth | undefined => {
   const keycloakInstance = useRef<KeycloakInstance>();
   const config = useConfig();
   const insights = useInsights();
 
-  if (config === undefined || insights.chrome.auth === undefined) {
+
+  if (insights.chrome.auth === undefined) {
     throw new Error("useAuth must be used inside a config provider, and insights auth");
   }
 
   const insightsChromeAuth = insights.chrome.auth;
 
-  const getKeycloakInstance = async () => {
+  const getKeycloakInstance = async (config: Config) => {
```

Comment:
this is really not right, you are inside a block where you called `useConfig`

Reply:
possibly improved by the latest change to ui-shared, please check.

Reply:
This is more for typescript. Because I removed the `config === undefined` now the config can be also undefined and the getKeycloakInstance requires a `Config`  and not `Config | undefined`

Reply:
Please update to the latest ui-shared, I recently resolved this problem there.

Reply:
I update the ui-shared but now if Config is undefined useConfig will fail so my only alternative is to create a fake Config while the config.json is fetched (only in the case the fetchConfig property is set to true), wouldn't this cause unpredictable errors?

Reply:
Right, you can't return a dummy config, this won't work.

The options are either to make the config load async (i.e. change Config to be more like `getConfig: ()=> Promise<Config> | undefined` to allow people to wait for the config to be loaded) or to have the system return a spinner whilst the config loads from the remote server.

I'd probably say the latter is fine, as it won't occur in production/staging where the config is compiled in.